rules/common_settings.bzl

```diff
@@ -12,50 +12,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Common build setting rules that return a BuildSettingInfo with the value of the build setting.
-For label-typed settings, use the native label_flag and label_setting rules."""
+"""Common build setting rules
 
-BuildSettingInfo = provider(fields = ["value"])
+These rules return a BuildSettingInfo with the value of the build setting.
+For label-typed settings, use the native label_flag and label_setting rules.
+
+More documentation on how to use build settings at
+https://docs.bazel.build/versions/master/skylark/config.html#user-defined-build-settings
+"""
+
+BuildSettingInfo = provider(
+    doc = """A singleton provider that contains the raw value of a build setting""",
+    fields = ["value"]
+)
 
 def _impl(ctx):
     return BuildSettingInfo(value = ctx.build_setting_value)
 
 int_flag = rule(
     implementation = _impl,
     build_setting = config.int(flag = True),
+    doc = """An int-typed build setting that is user settable""",
```

Comment:
"user-settable" is a bit confusing, how about "{may, may not} be set on the command line" ?

Reply:
Done.